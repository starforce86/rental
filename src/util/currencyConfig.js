// See: https://en.wikipedia.org/wiki/ISO_4217
// See: https://stripe.com/docs/currencies
export const subUnitDivisors = {
  EUR: 100,
  USD: 100,
  GBP: 100,
  AUD: 100,
  CAD: 100,
  DKK: 100,
  JPY: 100,
  SGD: 100,
  SEK: 100,
  CHF: 100,
  INR: 100,
  MXN: 100,
  CNY: 100,
  NOK: 100,
};
