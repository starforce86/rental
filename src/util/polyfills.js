//
// Use common polyfill dependencies from package.json
//

// Smoothscroll
require('smoothscroll-polyfill').polyfill();

// [].includes
require('array-includes').shim();

// [].find
require('array.prototype.find').shim();

// Object.entries
require('object.entries').shim();

// Object.values
require('object.values').shim();

// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/parseFloat
if (typeof Number.parseFloat === 'undefined' && typeof window !== 'undefined') {
  Number.parseFloat = window.parseFloat;
}

// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/parseInt
if (typeof Number.parseInt === 'undefined' && typeof window !== 'undefined') {
  Number.parseInt = window.parseInt;
}

// NaN is the only value in javascript which is not equal to itself.
// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN
if (typeof Number.isNaN === 'undefined') {
  // eslint-disable-next-line no-self-compare
  Number.isNaN = value => value !== value;
}
