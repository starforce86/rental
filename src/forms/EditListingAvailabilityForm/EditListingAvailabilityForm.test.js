// NOTE: renderdeep doesn't work due to map integration
import React from 'react';
import { renderShallow } from '../../util/test-helpers';
import { fakeIntl } from '../../util/test-data';
import { EditListingAvailabilityFormComponent } from './EditListingAvailabilityForm';

const noop = () => null;

describe('EditListingAvailabilityForm', () => {
  it('matches snapshot', () => {
    const tree = renderShallow(
      <EditListingAvailabilityFormComponent
        intl={fakeIntl}
        dispatch={noop}
        onSubmit={v => v}
        saveActionMsg="Save rules"
        updated={false}
        updateInProgress={false}
        availability={{
          calendar: {
            // '2018-12': {
            //   bookings: [],
            //   exceptions: [],
            //   fetchExceptionsError: null,
            //   fetchExceptionsInProgress: false,
            //   fetchBookingsError: null,
            //   fetchBookingsInProgress: false,
            // },
          },
          onCreateAvailabilityException: noop,
          onDeleteAvailabilityException: noop,
          onFetchAvailabilityExceptions: noop,
          onFetchBookings: noop,
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
