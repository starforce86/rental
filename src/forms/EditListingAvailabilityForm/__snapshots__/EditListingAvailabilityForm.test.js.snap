// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditListingAvailabilityForm matches snapshot 1`] = `
<ReactFinalForm
  availability={
    Object {
      "calendar": Object {},
      "onCreateAvailabilityException": [Function],
      "onDeleteAvailabilityException": [Function],
      "onFetchAvailabilityExceptions": [Function],
      "onFetchBookings": [Function],
    }
  }
  dispatch={[Function]}
  intl={
    Object {
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "now": [Function],
    }
  }
  onSubmit={[Function]}
  render={[Function]}
  saveActionMsg="Save rules"
  updateError={null}
  updateInProgress={false}
  updated={false}
/>
`;
